src/containers/Masthead/components/MastheadSearchForm.jsx

```diff
@@ -67,25 +70,50 @@ export class MastheadSearchForm extends Component {
     const splittedNdlaUrl = ndlaUrl.split(/\?/)[0].split('/');
 
     const urlId = splittedNdlaUrl[splittedNdlaUrl.length - 1];
-    if (!urlId.includes('urn:topic') && Number.isNaN(parseFloat(urlId))) return;
+
+    if (
+      !urlId.includes('urn:topic') &&
+      Number.isNaN(parseFloat(urlId)) &&
+      !splittedNdlaUrl.includes('subjects')
+    ) {
+      return;
+    }
 
     this.setState({ query: '' });
     if (urlId.includes('urn:topic')) {
-      fetchTopicArticle(urlId, locale).then(topicArticle => {
-        const arr = topicArticle.contentUri.split(':');
-        const id = arr[arr.length - 1];
-        history.push(toEditArticle(id, 'topic-article', locale));
-      });
-    } else if (
-      splittedNdlaUrl.includes('ndla.no') &&
-      splittedNdlaUrl.includes('node')
-    ) {
+      this.handleTopicUrl(urlId);
+    } else if (splittedNdlaUrl.includes('node')) {
       this.handleNodeId(urlId);
+    } else if (splittedNdlaUrl.includes('subjects')) {
+      this.handleFrontendUrl(ndlaUrl);
     } else {
       history.push(toEditArticle(urlId, 'standard', locale));
     }
   }
 
+  async handleTopicUrl(urlId) {
+    const { locale, history } = this.props;
+    const topicArticle = await fetchTopicArticle(urlId, locale);
+    const arr = topicArticle.contentUri.split(':');
+    const id = arr[arr.length - 1];
+    history.push(toEditArticle(id, 'topic-article', locale));
+  }
+
+  async handleFrontendUrl(url) {
+    const { locale, history } = this.props;
+    const splitted = url.split('subjects');
+    const taxonomyUrl = splitted[splitted.length - 1];
+    try {
+      const newArticle = await resolveUrls(taxonomyUrl, locale);
+      const arr = newArticle.contentUri.split(':');
```

Comment:
Vet at jeg er nok den skyldig for noen år siden her, men `arr` er et fryktelig dårlig navn. Kanskje kalt det `splittedUrl` eller noe sånt? Gjelder også i `handleTopicUrl`  Det samme gjelder `id`, burde vært `articleId`